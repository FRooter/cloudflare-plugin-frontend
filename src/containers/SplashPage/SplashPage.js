import React, { Component } from 'react';
import { connect } from 'react-redux';

import { FormattedMessage, injectIntl } from 'react-intl';
import { Box } from 'cf-component-box';
import { Button } from 'cf-component-button';
import { Heading } from 'cf-component-heading';
import { getConfigValue } from '../../selectors/config';

import { Card, CardSection } from 'cf-component-card';

import { LayoutContainer, LayoutRow, LayoutColumn } from 'cf-component-layout';

import BenefitsCollection from '../../containers/BenefitsCollection/BenefitsCollection';

import {
  CLOUDFLARE_SIGNUP_PAGE,
  LOGIN_PAGE,
  HOME_PAGE,
  SIGN_UP_PAGE
} from '../../constants/UrlPaths.js';
import { isLoggedIn } from '../../utils/Auth/Auth';
import { Link } from 'react-router';
import { push } from 'react-router-redux';

import { openWindow720x720 } from '../../utils/utils.js';

const cardBoxStyles = {
  margin: '2em auto',
  maxWidth: '1024px',
  textAlign: 'center'
};

const linkStyles = {
  textDecoration: 'none',
  color: '#1592E6'
};

const textStyles = {
  color: '#9A9D9E'
};

const cardPaddingStyles = {
  padding: '65px 0'
};

class SplashPage extends Component {
  constructor(props) {
    super(props);
    this.navigateToSignUpPage = this.navigateToSignUpPage.bind(this);
    this.openWindow720x720 = openWindow720x720.bind(this);
  }

  componentWillMount() {
    let { dispatch } = this.props;
    if (isLoggedIn()) {
      dispatch(push(HOME_PAGE));
    }
  }

  navigateToSignUpPage() {
    const { config } = this.props;
    let { dispatch } = this.props;
    const useHostAPILogin = getConfigValue(config, 'useHostAPILogin');

    if (useHostAPILogin) {
      dispatch(push(SIGN_UP_PAGE));
    } else {
      this.openWindow720x720(CLOUDFLARE_SIGNUP_PAGE);
    }
  }

  render() {
    const { config } = this.props;

    const integrationName = getConfigValue(config, 'integrationName');
    const integrationNameCapital =
      integrationName.charAt(0).toUpperCase() + integrationName.slice(1);

    return (
      <Box {...cardBoxStyles}>
        <Card>
          <Box {...cardPaddingStyles}>
            <CardSection>
              <LayoutContainer>
                <LayoutRow>
                  <LayoutColumn width={1 / 1}>
                    <Heading size={1}>
                      <FormattedMessage
                        id="container.splashPage.heading.speedUp"
                        values={{
                          integrationName: integrationNameCapital
                        }}
                      />
                    </Heading>
                  </LayoutColumn>
                </LayoutRow>

                <LayoutRow>
                  <LayoutColumn width={1 / 1}>
                    <Button
                      type="success"
                      onClick={() => this.navigateToSignUpPage()}
                    >
                      <FormattedMessage id="container.splashPage.button.createFreeAccount" />
                    </Button>
                  </LayoutColumn>
                </LayoutRow>

                <LayoutRow>
                  <LayoutColumn width={1 / 1}>
                    <p style={textStyles}>
                      <FormattedMessage id="container.splashPage.help.alreadyHaveAccount" />{' '}
                      <Link style={linkStyles} to={LOGIN_PAGE}>
                        <FormattedMessage id="container.splashPage.help.here" />
                      </Link>
                      .
                    </p>
                  </LayoutColumn>
                </LayoutRow>
                <LayoutRow>
                  <LayoutColumn width={1 / 1}>
                    <BenefitsCollection />
                  </LayoutColumn>
                </LayoutRow>
              </LayoutContainer>
            </CardSection>
          </Box>
        </Card>
      </Box>
    );
  }
}

function mapStateToProps(state) {
  return {
    config: state.config
  };
}

export default injectIntl(connect(mapStateToProps)(SplashPage));
