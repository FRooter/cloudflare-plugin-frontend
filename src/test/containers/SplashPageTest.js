import React from 'react';
import SplashPage from '../../containers/SplashPage/SplashPage';

import renderer from 'react-test-renderer';
import createMockStore from '../_helpers/createMockStore';

import { Provider } from 'react-redux';
import { StyleProvider } from 'cf-style-provider';
import { IntlProvider } from 'react-intl';

const messages = {
  'container.splashPage.heading.speedUp': 'Speed up and Optimize your WordPress Site with Cloudflare',
  'container.splashPage.help.alreadyHaveAccount': 'Have an account already? Sign in',
  'container.splashPage.help.here': 'here',
  'container.splashPage.button.createFreeAccount': 'Create Your Free Account',
  'component.benefitsFeature.globalCaching.title': 'Global Caching',
  'component.benefitsFeature.globalCaching.description': "Shorten your visitor's load times with content caching in our 110+ global locations.",
  'component.benefitsFeature.optimization.title': 'Website Optimization',
  'component.benefitsFeature.optimization.description': "Deliver content to users faster by enabling Cloudflare's content optimizations.",
  'component.benefitsFeature.security.title': 'Security',
  'component.benefitsFeature.security.description': 'Cloudflare offers protections against vulnerabilities including DDoS protection.',
  'component.benefitsFeature.insights.title': 'Insights',
  'component.benefitsFeature.insights.description': 'Monitor bandwidth saved, threats blocked, and more with built-in analytics.'
};

it('renders enabled correctly', () => {
  window.localStorage = {};
  const component = renderer
    .create(
      <Provider
        store={createMockStore({
          config: { config: { integrationName: 'wordpress' } }
        })}
      >
        <IntlProvider locale="en" messages={messages}>

          <StyleProvider
            cssNode={null}
            fontNode={null}
            dev={false}
            selectorPrefix="cf-"
          >
            <SplashPage />
          </StyleProvider>
        </IntlProvider>
      </Provider>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
