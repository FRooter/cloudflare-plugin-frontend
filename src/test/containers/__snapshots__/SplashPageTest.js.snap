// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders enabled correctly 1`] = `
<div
  className="cf-a cf-b cf-c"
>
  <section
    className="cf-card"
  >
    <div
      className="cf-d"
    >
      <div
        className="cf-card__section cf-card__section--default"
      >
        <div
          className="cf-layout__container"
        >
          <div
            className="cf-layout__row"
          >
            <div
              className="cf-layout__column"
              style={
                Object {
                  "width": "100.00%",
                }
              }
            >
              <h1
                className="cf-heading cf-heading--1"
              >
                <span>
                  Speed up and Optimize your WordPress Site with Cloudflare
                </span>
              </h1>
            </div>
          </div>
          <div
            className="cf-layout__row"
          >
            <div
              className="cf-layout__column"
              style={
                Object {
                  "width": "100.00%",
                }
              }
            >
              <button
                className="cf-btn cf-btn--success"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  Create Your Free Account
                </span>
              </button>
            </div>
          </div>
          <div
            className="cf-layout__row"
          >
            <div
              className="cf-layout__column"
              style={
                Object {
                  "width": "100.00%",
                }
              }
            >
              <p
                style={
                  Object {
                    "color": "#9A9D9E",
                  }
                }
              >
                <span>
                  Have an account already? Sign in
                </span>
                 
                <a
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "#1592E6",
                      "textDecoration": "none",
                    }
                  }
                >
                  <span>
                    here
                  </span>
                </a>
                .
              </p>
            </div>
          </div>
          <div
            className="cf-layout__row"
          >
            <div
              className="cf-layout__column"
              style={
                Object {
                  "width": "100.00%",
                }
              }
            >
              <div
                className="cf-layout__row"
              >
                <div
                  className="cf-flex"
                >
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "5.5556%",
                      }
                    }
                  >
                    Â 
                  </div>
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "22.222%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "padding": "30px 15px 30px 15px",
                        }
                      }
                    >
                      <img
                        src="assets/global-caching.svg"
                        style={
                          Object {
                            "display": "block",
                            "height": "auto",
                            "margin": "0 auto",
                            "maxWidth": "100px",
                            "width": "75px",
                          }
                        }
                      />
                      <span
                        href="#"
                        style={
                          Object {
                            "color": "#333333",
                            "display": "block",
                            "fontSize": "16px",
                            "padding": "20px 0 0 0",
                            "textAlign": "center",
                            "width": "100%",
                          }
                        }
                      >
                        Global Caching
                      </span>
                      <p
                        style={
                          Object {
                            "color": "#9A9D9E",
                            "fontSize": "12px",
                            "padding": "10px 0 0 0",
                            "textAlign": "center",
                          }
                        }
                      >
                        Shorten your visitor's load times with content caching in our 110+ global locations.
                      </p>
                    </div>
                  </div>
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "22.222%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "padding": "30px 15px 30px 15px",
                        }
                      }
                    >
                      <img
                        src="assets/web-optimization.svg"
                        style={
                          Object {
                            "display": "block",
                            "height": "auto",
                            "margin": "0 auto",
                            "maxWidth": "100px",
                            "width": "75px",
                          }
                        }
                      />
                      <span
                        href="#"
                        style={
                          Object {
                            "color": "#333333",
                            "display": "block",
                            "fontSize": "16px",
                            "padding": "20px 0 0 0",
                            "textAlign": "center",
                            "width": "100%",
                          }
                        }
                      >
                        Website Optimization
                      </span>
                      <p
                        style={
                          Object {
                            "color": "#9A9D9E",
                            "fontSize": "12px",
                            "padding": "10px 0 0 0",
                            "textAlign": "center",
                          }
                        }
                      >
                        Deliver content to users faster by enabling Cloudflare's content optimizations.
                      </p>
                    </div>
                  </div>
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "22.222%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "padding": "30px 15px 30px 15px",
                        }
                      }
                    >
                      <img
                        src="assets/security.svg"
                        style={
                          Object {
                            "display": "block",
                            "height": "auto",
                            "margin": "0 auto",
                            "maxWidth": "100px",
                            "width": "75px",
                          }
                        }
                      />
                      <span
                        href="#"
                        style={
                          Object {
                            "color": "#333333",
                            "display": "block",
                            "fontSize": "16px",
                            "padding": "20px 0 0 0",
                            "textAlign": "center",
                            "width": "100%",
                          }
                        }
                      >
                        Security
                      </span>
                      <p
                        style={
                          Object {
                            "color": "#9A9D9E",
                            "fontSize": "12px",
                            "padding": "10px 0 0 0",
                            "textAlign": "center",
                          }
                        }
                      >
                        Cloudflare offers protections against vulnerabilities including DDoS protection.
                      </p>
                    </div>
                  </div>
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "22.222%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "padding": "30px 15px 30px 15px",
                        }
                      }
                    >
                      <img
                        src="assets/insight.svg"
                        style={
                          Object {
                            "display": "block",
                            "height": "auto",
                            "margin": "0 auto",
                            "maxWidth": "100px",
                            "width": "75px",
                          }
                        }
                      />
                      <span
                        href="#"
                        style={
                          Object {
                            "color": "#333333",
                            "display": "block",
                            "fontSize": "16px",
                            "padding": "20px 0 0 0",
                            "textAlign": "center",
                            "width": "100%",
                          }
                        }
                      >
                        Insights
                      </span>
                      <p
                        style={
                          Object {
                            "color": "#9A9D9E",
                            "fontSize": "12px",
                            "padding": "10px 0 0 0",
                            "textAlign": "center",
                          }
                        }
                      >
                        Monitor bandwidth saved, threats blocked, and more with built-in analytics.
                      </p>
                    </div>
                  </div>
                  <div
                    className="cf-layout__column"
                    style={
                      Object {
                        "width": "5.5556%",
                      }
                    }
                  >
                    Â 
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
